### PR TITLE
Multiple Maximum Generation and Pumping for Hydro (Refactoring)

### PR DESCRIPTION
**Pull request by nikolaredstork**
This PR is created for clearing comments from RTE on feature/max-hydro-gen-pump-per-h-cr23 branch.
Link of the base PR :
https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1470

### CODE REVIEW

File: src/solver/hydro/management/management.h [hunk 1]
```diff
@@ -88,7 +88,7 @@ struct TmpDataByArea
 typedef struct
 {
     std::vector<double> HydrauliqueModulableQuotidien; /* indice par jour */
-    std::vector<double> NiveauxReservoirsDebutJours;   //Niveaux (quotidiens) du reservoir de d�but
+    std::vector<double> NiveauxReservoirsDebutJours;   //Niveaux (quotidiens) du reservoir de d�but
```
Comment (flomnes) [inline src/solver/hydro/management/management.h]:
```suggestion
    std::vector<double> NiveauxReservoirsDebutJours;   //Niveaux (quotidiens) du reservoir de début
```

File: src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.h [hunk 2]
```diff
@@ -54,17 +58,19 @@ class DataTransfer
...
-
-
+private:
+    bool LoadDailyMaxPowersAndEnergies(const AnyString& folder, const Area& area);
+    bool SaveDailyMaxEnergy(const AnyString& folder, Area& area);
+    bool SaveDailyMaxPowerAsHourly(const AnyString& folder, Area& area);
+    bool SaveMaxGenerationEnergy(const AnyString& folder, Area& area);
+    bool SaveMaxPumpingEnergy(const AnyString& folder, Area& area);
+    bool SaveDailyMaxGenPowerAsHourly(const AnyString& folder, Area& area);
+    bool SaveDailyMaxPumpPowerAsHourly(const AnyString& folder, Area& area);
```
Comment (flomnes) [inline src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.h]: Add a comment about legacy ? For versions newer than XXX, use YYY
For older versions, use ZZZ

File: src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.cpp [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+void TransferDailyMaxPowerAsHourly(Matrix<double, int32_t>::ColumnType& hourlyColumn,
+                                    const Matrix<double>::ColumnType& dailyColumn)
+{
+    uint hours = 0;
+    uint days = 0;
+
+    while (hours < HOURS_PER_YEAR && days < DAYS_PER_YEAR)
+    {
+        for (uint i = 0; i < 24; ++i)
```
Comment (flomnes) [inline src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.cpp]:
```suggestion
        for (uint i = 0; i < HOURS_PER_DAY; ++i)
```
?

File: src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.h (hunk 2 above)
Comment (nikolaredstork) [reply src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.h]: All of these functions are used when loading old study! HydroMaxTimeSeriesReader class is just for supporting old studies.

File: src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.cpp (hunk 3 above)
Comment (nikolaredstork) [reply src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.cpp]: I've implemented that macro earlier today when i was refactoring AreaScratchpad class, but I did not pushed it. Regarding AreaScratchpad class will you be able to give me your comment on this?
https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1470#discussion_r1333034108

File: src/solver/hydro/management/management.h (hunk 1 above)
Comment (nikolaredstork) [reply src/solver/hydro/management/management.h]: Done!

File: src/libs/antares/study/parts/hydro/series.h [hunk 4]
```diff
@@ -153,25 +154,103 @@ class DataSeriesHydro
...
+    public:
+        TsActions(Matrix<double, int32_t>& maxHourlyGenPower_,
+                  Matrix<double, int32_t>& maxHourlyPumpPower_);
+
+        void handleBothZeros(const AreaName& areaID);
+        [[noreturn]] void handleBothGreaterThanOne(const AreaName& areaID) const;
+        void resizeWhenOneTS(Area& area, uint nbTimeSeriesSup_);
+
+    private:
+        Matrix<double, int32_t>& maxHourlyGenPower;
```
Comment (guilpier-code) [inline src/libs/antares/study/parts/hydro/series.h]: Please keep the trailing underscores for private variables, not for methods arguments.

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/tests/src/libs/antares/study/parts/hydro/CMakeLists.txt [hunk 5]
```diff
@@ -1,76 +1,66 @@
 # Useful variables definitions
 set(src_libs_antares_study "${CMAKE_SOURCE_DIR}/libs/antares/study")
 set(src_libs_antares "${CMAKE_SOURCE_DIR}/libs/antares")
+set(src_test_utilis "${CMAKE_SOURCE_DIR}/tests/src/utlis")
```
Comment (flomnes) [inline src/tests/src/libs/antares/study/parts/hydro/CMakeLists.txt]: I think there are two typos
```suggestion
set(src_test_utils "${CMAKE_SOURCE_DIR}/tests/src/utils")
```

Comment (nikolaredstork) [reply src/tests/src/libs/antares/study/parts/hydro/CMakeLists.txt]: Done!

File: src/libs/antares/study/parts/hydro/series.h (hunk 4 above)
Comment (nikolaredstork) [reply src/libs/antares/study/parts/hydro/series.h]: Done!

**Pull request merged**